### ISSUE TITLE
Send filters configuration in the link to allow exact configuration share

### ISSUE THREAD

**Issue by omerxx** [hasadna/anyway#423]
The current state (after implementing the new filter bar) only allows date ranges, zoom level and geo location to be shared in the link.

Instead of concatenating the 20+ params that would have created a very long address, we should generate a unique code for every specific filter configuration.
The code would be parsed and sent to `bounding_box_query`.
(URL parameters are being sent in `getCurrentUrlParams` in `app.js`)

**Comment by omerxx** [comment]
From what I understand the problem is with the new filter bar design.
It does not communicate with the backbone model and parameters are not conveyed in a way the URL can manipulate.

We should redefine the variables using `app.model.set...`, and then `bind` every filter bar function with the model.
We should also have URL parameters in `getCurrentUrlParams()` and in `buildMarkerParams()` set using `app.model.get()` rather than `this.<parameter>`

I'd need some backbone expertise help here... @aviklai are you up for it?

**Comment by omerxx** [comment]
@aviklai The hash string shouldn't have a map of every single combination.
There are a little over 20 filtering options, so we should have a 20+ characters string, where each char is a symbolic integer of a single filter. When the app reads the string it would go through this list and make sure each option is following the required config. 
In face this is not an actual 'hash' table but a way to squeeze 20+ url variables into a single 1.

**Comment by danielhers** [comment]
Some filters (like date) don't fit in a single character. But we can use
some Huffman coding or something...

**Comment by omerxx** [comment]
No need for that. Date and geo location is already part of the existing URL. What I'd like to "hash" is the rest of the new filter options

**Comment by danielhers** [comment]
Right, but shorter links would be beneficial. That's why the idea of
hashing the URL is attractive.

**Comment by omerxx** [comment]
I agree, but I think when we discuss date ranges readability, and edit-ability of the URL is important just as well

**Comment by danielhers** [comment]
Right. We could provide an optional short URL like in Google Maps, when you
share a map.

**Comment by aviklai** [comment]
I am working on it. Basically, what I am doing is making a link that will have all the parameters and when you use it, it will load all the relevant data and than change the URL to a parameterless one.
